### ISSUE TITLE
Support standalone plugins in PL generator meta tag

### ISSUE THREAD

**Issue by felixarntz** [WordPress/performance#692]
The Performance Lab plugin has had a generator meta tag since shortly after its launch (see #321). Now that we are publishing the modules as standalone plugins and eventually will break them out, we need to also iterate on how that generator tag includes the relevant information. We would still want to make sure the generator tag includes which of the modules or equivalent standalone plugins the site uses.

I can see two alternative approaches for this:
1. Either we include the module name in the generator tag like we do today, even if it's actually the module's equivalent standalone plugin that is being used.
2. Or we keep the existing generator tag as is, only for modules, and introduce a second generator tag that captures any of the standalone plugins including their versions.

On the one hand, approach 1. is simpler and ties in seamlessly with how the generator tag works today. Any tooling that makes use of the data to measure adoption (using the public HTTP Archive dataset) will continue to function without modifications.

On the other hand, approach 2. would clearly separate which features are used as modules vs which features are used as plugins, and it could also include each standalone plugin's version number. This provides more information, but is also a bit more effort and might be unnecessarily verbose.

Do we actually need the additional information that approach 2. provides? If yes, I think we should go with it. If not, I think we should go with approach 1.

**Comment by joemcgill** [comment]
Good thinking, @felixarntz. It seems to me like we should plan to fully decouple the standalone plugins from the Performance Lab plugin, so I'd lean towards option number two.

**Comment by mukeshpanchal27** [comment]
@felixarntz I agree with @joemcgill's point, so I also go with option two.

**Comment by felixarntz** [comment]
Thanks @joemcgill @mukeshpanchal27, let's go with that approach then, i.e. a 2nd generator meta tag.

Maybe it could look like this:
```html
<meta name="generator" content="Performance Lab plugins: sqlite-database-integration:2.0, webp-uploads:1.0.0">
```

It would basically list all standalone plugins associated with the Performance Team / Performance Lab repository that are active, including their individual version number. WDYT?

**Comment by mukeshpanchal27** [comment]
Yes make sense. So this new generator tag will be added as a standalone plugin or in the Performance Lab plugin?

**Comment by felixarntz** [comment]
@mukeshpanchal27 I think we would add it in the Performance Lab plugin, so that it's central. Otherwise, it's gonna be a maintenance nightmare to replicate and ensure it is in every standalone plugin.

**Comment by joemcgill** [comment]
I think I misunderstood what you were proposing previously. I had assumed that each standalone plugin would generate its own individual generator tag rather than making it the responsibility of the performance lab plugin to do so. However I'm now understanding the proposal as being a second tag (i.e., not individual tags) generated by the Performance Lab plugin that displays all of the standalone plugins related to the performance lab that are active on the site. Is that what you're proposing? If so, I don't really see the value in separating this into two tags at all.

My preference would be that each standalone plugin would be responsible for its own tag and that it's not to be dependent on whether the Performance Lab plugin is installed or not.

**Comment by felixarntz** [comment]
@joemcgill 
> My preference would be that each standalone plugin would be responsible for its own tag and that it's not to be dependent on whether the Performance Lab plugin is installed or not.

This would be another option to consider, conceptually I like that idea.

It has a few technical caveats though, we'll need to duplicate that code into every plugin (which is probably fine because it's little, but still worth noting as a tradeoff). It'll also mean we'll have to put logic there to make sure it is _not_ output when used as a module (relevant for the short-term stage where we'll have modules coexist with their standalone plugins). We can certainly do that, it's just a bit clunky, e.g. we can check if the module's/plugin's version constant starts with "Performance Lab", we know it's a module and should bail early in the function to output the generator tag.

**Comment by mukeshpanchal27** [comment]
We have three options here for the issue. If we go with third option than also we can merged all standalone plugin in single generator tag using action/filter. I will async check if it works or not.

`<meta name="generator" content="Performance Lab plugins: sqlite-database-integration:2.0, webp-uploads:1.0.0">`

**Comment by felixarntz** [comment]
@mukeshpanchal27 I think if we want to put the generator tags in the standalone plugins, then we should also make them completely standalone, no need then to have it centralized via PL and only work if that plugin is active. The per-plugin route is probably the most reliable in terms of adoption and version reporting, since it doesn't require PL to be active as well.

**Comment by mukeshpanchal27** [comment]
Closed in #696
